src/Validation/Rule/ExecutableDefinitionRule.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Digia\GraphQL\Validation\Rule;
+
+use Digia\GraphQL\Error\ValidationException;
+use Digia\GraphQL\Language\AST\Node\DocumentNode;
+use Digia\GraphQL\Language\AST\Node\ExecutableDefinitionNodeInterface;
+use Digia\GraphQL\Language\AST\Node\NamedTypeNode;
+use Digia\GraphQL\Language\AST\Node\NodeInterface;
+use Digia\GraphQL\Language\AST\Node\SchemaDefinitionNode;
+use function Digia\GraphQL\Validation\nonExecutableDefinitionMessage;
+
+/**
+ * Executable definitions
+ *
+ * A GraphQL document is only valid for execution if all definitions are either
+ * operation or fragment definitions.
+ */
+class ExecutableDefinitionRule extends AbstractRule
+{
+    /**
+     * @inheritdoc
+     */
+    public function enterNode(NodeInterface $node): ?NodeInterface
+    {
+        if ($node instanceof DocumentNode) {
```

Comment:
I think it is easier to read if you return soon, return often :P

Reply:
I agree. I'll make this change.

Reply:
Actually no, because we always want to return `$node`.